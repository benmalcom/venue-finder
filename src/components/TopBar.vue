<template>
    <div class="container-fluid top-bar">
        <div class="row col-md-9 mx-auto top-bar-inner">
            <div class="col-md-6 d-flex align-items-center">
                <h6 class="text-left text-white">VENUE <strong>FINDER</strong></h6>
            </div>
            <div class="offset-md-3 col-md-3 mt-3 d-flex align-items-center">
                <div class="media">
                    <label class="mr-2 text-white"><i class="fa fa-map-marker fa-2x"></i></label>
                    <div class="media-body text-white">
                        <h5 class="text-sm mb-0">Cras purus odio, vestibulum in vulputate.</h5>
                        <p class="text-xs current-location mt-0">Current location</p>
                    </div>
                </div>
            </div>
        </div>
    </div>
</template>

<script>
	export default {
		name: 'TopBar',
        props: {

        },
        data: function () {
            return {

            }
        }
	}
</script>

<style scoped>

</style>
