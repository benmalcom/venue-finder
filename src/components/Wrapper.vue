<template>
    <!-- wrapper template -->
    <div class="wrapper flex-grow">
        <slot></slot>
    </div>
</template>


<script>
	export default {
		name: 'Wrapper'
	}
</script>

<style scoped>

</style>
