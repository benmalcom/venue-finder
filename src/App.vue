<template>
    <Wrapper>
        <vue-progress-bar></vue-progress-bar>
        <TopBar :position="currentPosition"></TopBar>
        <FilterBox :categories="venueCategories" @search-filter="onSearchFilter"></FilterBox>
        <VenueList v-if="currentPosition" :venues="venues"></VenueList>
        <div class="alert alert-warning col-md-9 mx-auto mt-30" v-if="!currentPosition && loading">
            Trying to access your location...
        </div>
        <div class="alert alert-info col-md-9 mx-auto mt-30" v-else-if="currentPosition && loading">
            Getting venues...
        </div>
        <div class="alert alert-danger col-md-9 mx-auto mt-30" v-else-if="!currentPosition && !loading">
            Unable display venues, please refresh your browser if this takes long
        </div>
    </Wrapper>
</template>
<style lang="scss">
    @import '~bootstrap/scss/bootstrap.scss';
    $fa-font-path: '~font-awesome/fonts';
    @import '~font-awesome/scss/font-awesome.scss';
    @import './assets/css/app.scss';
</style>
<script>
	import TopBar from './components/TopBar.vue';
    import FilterBox from './components/FilterBox.vue';
    import VenueList from './components/venue/VenueList.vue';
    import Wrapper from './components/Wrapper.vue';
    import axios from './axios';
    import { getCurrentPosition } from './config';
	export default {
		name: 'App',
		components: {FilterBox, TopBar, VenueList, Wrapper},
		mounted: function () {
			$('[data-toggle=offcanvas]').click(function () {
				$('.offcanvas').toggleClass('active');
			});
			$('.close-filter').click(function () {
				$('.offcanvas').toggleClass('active');
			});
		},
        created: async function() {
	        try{
		        this.$Progress.start();
		        this.loading = true;
		        const position = await getCurrentPosition();
		        const  {latitude, longitude} = position.coords;
		        this.currentPosition =  {latitude, longitude};


		        let response = await axios.get('/search', { params: {
				        ll: `${latitude},${longitude}`}
		        });
		        this.venues = response.venues;

		        response = await axios.get('/categories');
		        this.$Progress.finish();
		        this.loading = false;
		        const { categories } = response;
		        console.log('categories ', categories);
		        this.venueCategories = categories.map(({id, shortName}) => ({id, shortName}));

	        } catch (e) {
		        this.$Progress.fail();
		        this.loading = false;
		        console.log('error ', e.message);
	        }
        },
		data() {
			return {
				currentPosition: null,
				loading: false,
				searchFilter: {},
				venueCategories: [],
				venues: []
			};
		},
        methods: {
	        onSearchFilter: async function(data) {
		        Object.assign(this.searchFilter, data);
		        try{
			        this.$Progress.start();
			        const  {latitude, longitude} = this.currentPosition;
			        const response = await axios.get('/search', { params: {
			        	    ll: `${latitude},${longitude}`, ...data
			            }
			        });
			        this.$Progress.finish();
			        this.venues = [...response.venues];
		        } catch (e) {
			        this.$Progress.fail();
			        console.log('error ', e.message);
		        }
	        }
        }
	};
</script>
