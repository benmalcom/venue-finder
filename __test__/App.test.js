import { shallow, mount } from 'vue-test-utils';
import App from '../src/App.vue';
import TopBar from '../src/components/TopBar.vue';
import Wrapper from '../src/components/Wrapper.vue';

describe('Component: App.vue', () => {
	let comp;

	beforeEach(() => {
		comp = shallow(App);
	});

	it('Has the expected html structure', () => {
		expect(comp.element).toMatchSnapshot();
	});


	it('Should contain Wrapper component', () => {
		expect(comp.find(Wrapper).exists()).toBeTruthy();
	});

	it('Should set component default data', () => {
		expect(comp.vm.currentPosition).toBeNull();
		expect(comp.vm.venues).toBeNull();
		expect(Array.isArray(comp.vm.venueCategories)).toBeTruthy();
		expect(comp.vm.message).toBeDefined();
		expect(typeof comp.vm.message).toEqual("string");
		expect(comp.vm.alertClass).toBeDefined();
		expect(typeof comp.vm.alertClass).toEqual("string");
		expect(typeof comp.vm.searchFilter).toEqual("object");
	});
});
