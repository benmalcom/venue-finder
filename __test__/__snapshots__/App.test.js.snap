// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: App.vue Has the expected html structure 1`] = `<!---->`;
